sys/include/cbmplex.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2018 Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup   sys_cbmplex
+ * @{
+ *
+ * @file
+ * @brief   cbmplex interface definitions
+ *
+ * @author  Matthew Blue <matthew.blue.neuro@gmail.com>
+ */
+
+#ifndef CBMPLEX_H
+#define CBMPLEX_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   cbmplex callback type
+ */
+typedef void (*cbmplex_cb_t)(void *);
+
+/**
+ * @brief   cbmplex list entry structure
+ */
+typedef struct cbmplex {
+    struct cbmplex *next;
+    uint8_t flags;
+    uint8_t cbid;
```

Comment:
What is the idea behind the callback ID?

Reply:
This is whatever the system using the multiplexer uses to identify the callback. In the case of pin interrupts, it would be the pin number.

Reply:
Not sure I understand what you mean. How does the "system" here know the `cbid`? It is not passed as argument in the callback.

Reply:
See example.